templates/zerver/development/design_testing/buttons.html

```diff
@@ -0,0 +1,171 @@
+                <i class="zulip-icon zulip-icon-move-alt"></i>
+                <span class="action-button__label">Button joy</span>
+            </button>
+        </div>
+    </section>
+    <section>
+        <div class="section-heading">Danger Buttons</div>
+        <div class="action-button-group">
+            <button class="action-button action-button--primary-danger" tabindex=0>
+                <i class="zulip-icon zulip-icon-move-alt"></i>
+                <span class="action-button__label">Button joy</span>
+            </button>
+            <button class="action-button action-button--secondary-danger" tabindex=0>
+                <i class="zulip-icon zulip-icon-move-alt"></i>
+                <span class="action-button__label">Button joy</span>
+            </button>
+            <button class="action-button action-button--tertiary-danger" tabindex=0>
+                <i class="zulip-icon zulip-icon-move-alt"></i>
+                <span class="action-button__label">Button joy</span>
+            </button>
+        </div>
+    </section>
+    <section>
+        <div class="section-heading">Controls</div>
+        <div class="design-testing-controls">
+            <div class="design-testing-control">
+                <span class="control-label">Dark Theme</span>
+                <div role="group" class="tab-picker">
+                    <input type="radio" id="enable_dark_theme" class="tab-option" name="dark-theme-select"/>
+                    <label role="menuitemradio" class="tab-option-content" for="enable_dark_theme" tabindex="0">
+                        <span>Enable</span>
+                    </label>
+                    <input type="radio" id="disable_dark_theme" class="tab-option" name="dark-theme-select" checked/>
+                    <label role="menuitemradio" class="tab-option-content" for="disable_dark_theme" tabindex="0">
+                        <span>Disable</span>
+                    </label>
+                    <span class="slider"></span>
+                </div>
+            </div>
+            <div class="design-testing-control">
+                <span class="control-label">Button Icon</span>
+                <div role="group" class="tab-picker">
+                    <input type="radio" id="enable_button_icon" class="tab-option" name="button-icon-select" checked/>
+                    <label role="menuitemradio" class="tab-option-content" for="enable_button_icon" tabindex="0">
+                        <span>Enable</span>
+                    </label>
+                    <input type="radio" id="disable_button_icon" class="tab-option" name="button-icon-select"/>
+                    <label role="menuitemradio" class="tab-option-content" for="disable_button_icon" tabindex="0">
+                        <span>Disable</span>
+                    </label>
+                    <span class="slider"></span>
+                </div>
+            </div>
```

Comment:
Do we need a control for the background color for this panel, i.e. the background behind the button? I'm not sure, but I think we probably should at least make sure the buttons appear on the standard settings background, ideally using a shared CSS variable to make sure it's always correct. And if we have multiple common backgrounds behind buttons (say, settings vs. modal are different), then it would probably be good to have a dropdown for which of those to use.

Reply:
That actually sounds like a great idea, since I'm already including `app_variables.css` in the `devtools/buttons` page, and it will help visualize the contrast between the buttons and backdrops.

Reply:
Added a background selector dropdown which syncs with the variables in use in `app_variables.css`.
![Screenshot 2024-12-03 at 11 46 04 PM](https://github.com/user-attachments/assets/39b75405-ff56-471e-853e-7a45032e4ac1)